### ISSUE TITLE
Replace Upstream Test Code With Dependency

### ISSUE THREAD

**Issue by nicolaiparlog** [junit-pioneer/junit-pioneer#6]
#4 enabled tests to be written like in JUnit 5. Part of the implementation copy-pasted a bunch of classes from the project (see f613211 for reasons). As soon as JUnit publishes test artifacts (see junit-team/junit5#572) a proper dependency can be established and the class can be removed.

**Comment by Bukama** [comment]
> As soon as JUnit publishes test artifacts (see [junit-team/junit5#572](https://github.com/junit-team/junit5/issues/572)) a proper dependency can be established and the class can be removed.

I asked for an update yesterday and Marc replied as this

> Please upgrade to 5.4.2 or later and use junit-platform-testkit.

So we can remove the classes

**Comment by nicolaiparlog** [comment]
Great, lets do that!

**Comment by nicolaiparlog** [comment]
And thanks for following up on this! :+1:

**Comment by nicolaiparlog** [comment]
I realize right now that I never considered whether Marc's answer applies to what we specifically are doing with the internal API. Did you give this a try? Maybe we should. :) If you didn't and don't want to, we can do it tomorrow on stream.

**Comment by nicolaiparlog** [comment]
We gave this a try and it looks like the API does everything we need. I'll merge #210 momentarily and then we can get to work on this. I created a proof of concept in [issues/6-testkit-api](https://github.com/junit-pioneer/junit-pioneer/tree/issues/6-testkit-api). Let's do it as follows:

* replace all uses of the internal API with the new one
* open a PR, so we can have a look and see how we can improve by adding our own functionality
* write that functionality and refactor accordingly (same branch)
* another review, then we merge
* open an issue with JUnit that our API rocks harder than theirs :grin:
